### ISSUE TITLE
Spartan input dialog: select default text

### ISSUE THREAD

**Issue by stacksmith** [skyjake/lagrange#490]
TLDR: I suggest a tiny change: select the default text in the input field, allowing easy replacement (by default).  

# Current behaviour
The `=:` input dialog uses the text following the `?` of the url as default text in the entry field.  If the default is accepted, the server will receive a duplicate of the text following the ?.  For example,
```
=: spartan://tilde.cafe/~spellbinding/game?session-id-1234 Enter a word...
```
will display an input box, containing the following default text: `session-id-1234`, and accepting it, the server will be hit with `spartan://tilde.cafe/~spellbinding/game?session-id-1234?session-id-1234`.

However, in most cases, the user will want to replace the default with their input.  If the text starts selected, just typing new input will replace it instead of appending it to the default.

# Passing state as session-ids
First of all, thank you for resisting the urge to strip the data following the `?` in the prompt URL.  Since the Spartan spec does not provide for any way to hide session-ids or other data in the prompt, it seems that hanging session data off the '?` in the URL is **the only way to pass state around.**

Your decision to use the prompt's URL query data as prompt default is reasonable (although sub-optimal in my case - I would love to have it be hidden in the input field but sent along with the user input...).

Whatever you do, please retain this feature (not trimming the existing query from the prompt URL prior to appending user input) - it is the only way to pass the application state to the user and back!

# Select the default text
Since the default (in my case, session-id) is not selected, if the user just types a word, it will be appended to the default text.  This is pretty much never what you want - you either accept the default as is, or more generally, replace it.  Starting with it selected seems more intuitively correct - you can just click on accept, or type in new data.

In the rare case of re-editing something, you will need to use arrow-keys to position the cursor (which you have to do no matter what) or click with the mouse.  

I can't imagine the case for appending the default text, which is what the current scenario is.

**Comment by skyjake** [comment]
>  server will receive a duplicate of the text following the ?

This was a bug and has now been fixed.

With the prompt URL contains a query string, that is used as the default input in the UI and is now selected, allowing easy replacement.

**Comment by stacksmith** [comment]
Just to be clear, the original query (session-id in my case) will still be sent back, right?  

The bug you are referring to is not highlighting the text?
